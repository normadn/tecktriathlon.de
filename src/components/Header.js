import React, { Component } from "react";

import ContentWrapper from "./ContentWrapper";
import MobileMenu from "./MobileMenu";
import Menu from "./Menu";

let prevScrollY = 0;

const menuItems = [
  "wertungen",
  "sponsoren",
  "informationen",
  "strecken",
];

class Header extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isHeaderVisible: true,
      isMobileMenuVisible: false,
      isLogoVisible: false,
      isAtTheTop: true,
      standaloneVersion: props.standaloneVersion || false
    };
  }

  componentDidMount() {
    window.onscroll = () => {
      if (!this.props.standaloneVersion) {
        const isAtTheTop = window.scrollY === 0;
        const isLogoVisible = window.scrollY > 200;
        const isHeaderVisible =
          window.scrollY < prevScrollY || window.scrollY < 100;
        if (
          this.state.isHeaderVisible !== isHeaderVisible ||
          this.state.isLogoVisible !== isLogoVisible ||
          this.state.isAtTheTop !== isAtTheTop
        ) {
          this.setState({ isHeaderVisible, isLogoVisible, isAtTheTop });
        }
        prevScrollY = window.scrollY;
      }
    };
  }

  disableScrolling = (e) => e.preventDefault();

  enableScrolling = () => true;

  onMobileMenuClicked = () => {
    this.setState({ isMobileMenuVisible: !this.state.isMobileMenuVisible });
    document.body.style.overflow = this.state.isMobileMenuVisible
      ? ""
      : "hidden";
  };

  render() {
    // Throws an error during build command
    try {
      document.ontouchmove = this.state.isMobileMenuVisible
        ? this.disableScrolling
        : this.enableScrolling;
    } catch (e) {}

    return (
      <div>
        <div
          css={{
            width: "100%",
            position: "fixed",
            top: this.state.isHeaderVisible ? "0px" : "-100px",
            transition: "0.5s",
            backgroundColor: "#ffffff",
            opacity: this.state.isMobileMenuVisible
              ? "1.0"
              : (this.state.isHeaderVisible && "0.95") || "0",
            zIndex: "5",
            "@media(min-width:860px)": {
              backgroundColor: this.props.standaloneVersion
                ? "#ffffff"
                : this.state.isAtTheTop
                  ? "rgba(0,0,0,0)"
                  : "#ffffff"
            }
          }}
        >
        <ContentWrapper>
          <div
            css={{
              "@media(max-width:860px)": {
                display: "grid",
                gridTemplateColumns: "auto auto"
              }
            }}
          >
            <Menu
              menuItems={this.props.menuItems || menuItems}
              imgCss={true}
              standaloneVersion={this.props.standaloneVersion}
            />
            <MobileMenu
              menuItems={this.props.menuItems || menuItems}
              isMenuVisible={this.state.isMobileMenuVisible}
              toggleMenu={this.onMobileMenuClicked}
            />
          </div>
        </ContentWrapper>
      </div>
      </div>
    );
  }
}

export default Header;
