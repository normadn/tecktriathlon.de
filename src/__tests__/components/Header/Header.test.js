import React from "react";
import Header from "../../../components/Header";
import { shallow } from "enzyme";

const menuItems = ["b1", "b2", "b3"];
describe("Header", () => {
  it("renders correctly", () => {
    expect(shallow(<Header menuItems={menuItems} />)).toMatchSnapshot();
  });
});
