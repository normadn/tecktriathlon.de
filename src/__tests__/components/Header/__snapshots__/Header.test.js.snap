// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div>
  <div
    css={
      Object {
        "@media(min-width:860px)": Object {
          "backgroundColor": "rgba(0,0,0,0)",
        },
        "backgroundColor": "#ffffff",
        "opacity": "0.95",
        "position": "fixed",
        "top": "0px",
        "transition": "0.5s",
        "width": "100%",
        "zIndex": "5",
      }
    }
    data-cy="mlh-banner"
  >
    <_default>
      <div
        css={
          Object {
            "@media(max-width:860px)": Object {
              "display": "grid",
              "gridTemplateColumns": "auto auto",
            },
          }
        }
      >
        <Menu
          imgCss={false}
          menuItems={
            Array [
              "b1",
              "b2",
              "b3",
            ]
          }
        />
        <MobileMenu
          isMenuVisible={false}
          menuItems={
            Array [
              "b1",
              "b2",
              "b3",
            ]
          }
          toggleMenu={[Function]}
        />
      </div>
    </_default>
  </div>
</div>
`;
